File: sort/rainbowSort.java
//0000 left 111 i??? right 2222
// 3 numbers 
public void rainbowSort(int[] arr){
	if(arr == null || arr.length <= 1)
		return;
	int left = 0;
	int right = arr.length -1;
	int i = 0;
	while(i <= right){
		if(arr[i] == 0){
			swap(arr, left++, i++);
			// left++;
			// i++;
		}else if(arr[i] == 1){
			i++;
		}else{
			swap(arr, right--, i);
			//right--;
		}
	}
}

File: sort/rainbowSort.py
def rainbowsort(self, nums:List[int]) -> None:
    """
    input: int[] array
    return: int[]
    """
    # write your solution here
    if not array:
      return array

    left = 0
    i = 0
    right = len(array)-1

    while i <= right:
      if array[i] == -1:
        array[i], array[left] = array[left], array[i]
		# self.swap(array, i, left)
        i+=1
        left+=1
      elif array[i] == 0:
        i+=1
      else:
        array[i], array[right] = array[right], array[i]
        right-=1
    return array

def swao(self, arr, i, left):
	array[i], array[left] = array[left], array[i]
